lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -32,43 +32,31 @@ def initialize(app, options = {})
           @app = app
         end
 
-        def configure
-          # ensure that the configuration is executed only once
-          return clean_context if @tracer && @service
-
-          # retrieve the current tracer and service
-          @tracer = Datadog.configuration[:rack][:tracer]
-          @service = Datadog.configuration[:rack][:service_name]
-          @distributed_tracing = Datadog.configuration[:rack][:distributed_tracing]
-        end
-
-        # rubocop:disable Metrics/MethodLength
         def call(env)
-          # configure the Rack middleware once
-          configure()
+          # retrieve integration settings
+          tracer = Datadog.configuration[:rack][:tracer]
+          service = Datadog.configuration[:rack][:service_name]
+          distributed_tracing = Datadog.configuration[:rack][:distributed_tracing]
 
           trace_options = {
-            service: @service,
+            service: service,
             resource: nil,
             span_type: Datadog::Ext::HTTP::TYPE
           }
 
-          request_span = nil
-          begin
-            if @distributed_tracing
-              context = HTTPPropagator.extract(env)
-              @tracer.provider.context = context if context.trace_id
-            end
-          ensure
-            # start a new request span and attach it to the current Rack environment;
-            # we must ensure that the span `resource` is set later
-            request_span = @tracer.trace('rack.request', trace_options)
+          if distributed_tracing
+            context = HTTPPropagator.extract(env)
+            tracer.provider.context = context if context.trace_id
           end
 
+          # start a new request span and attach it to the current Rack environment;
+          # we must ensure that the span `resource` is set later
+          request_span = tracer.trace('rack.request', trace_options)
           env[:datadog_rack_request_span] = request_span
 
           # call the rest of the stack
           status, headers, response = @app.call(env)
+          [status, headers, response]
```

Comment:
`ensure` doesn't change the return value of the function, so this should be here in the function body